tests/nnstreamer_filter_shared_model/unittest_filter_shared_model.cc

```diff
@@ -0,0 +1,231 @@
+
+  *str = g_strdup_printf (
+      "filesrc location=%s ! pngdec ! videoscale ! imagefreeze ! videoconvert ! "
+      "video/x-raw,format=RGB,framerate=10/1 ! tensor_converter ! tee name=t t. ! "
+      "queue ! tensor_filter name=filter1 framework=tensorflow-lite model=%s is-updatable=TRUE "
+      "shared-tensor-filter-key=%s ! tensor_sink name=sink1 t. ! "
+      "queue ! tensor_filter name=filter2 framework=tensorflow-lite model=%s is-updatable=TRUE "
+      "shared-tensor-filter-key=%s ! tensor_sink name=sink2",
+      image_path, model_path1, shared_key, model_path2, shared_key);
+  g_free (root_path);
+  g_free (model_path1);
+  g_free (model_path2);
+  g_free (image_path);
+}
+
+/**
+ * @brief Test filters share key but have different model paths.
+ */
+TEST (nnstreamerFilterSharedModel, tfliteSharedModelNotEqual_n)
+{
+  gchar *pipeline_str;
+  GstElement *pipeline;
+  _get_pipeline_str (&pipeline_str, model_name1, model_name2);
+  pipeline = gst_parse_launch (pipeline_str, NULL);
+
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
+  g_usleep (TEST_DEFAULT_SLEEP_TIME);
+  EXPECT_EQ (setPipelineStateSync (pipeline, GST_STATE_NULL, UNITTEST_STATECHANGE_TIMEOUT), 0);
+
+  g_free (pipeline_str);
+  gst_object_unref (pipeline);
+}
+
+/**
+ * @brief Test filter has invalid shape for shared model
+ */
+TEST (nnstreamerFilterSharedModel, tfliteInvalidShape_n)
+{
+  const gchar *src_root = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+  gchar *root_path = src_root ? g_strdup (src_root) : g_get_current_dir ();
+  gchar *model_path1 = g_build_filename (
+      root_path, "tests", "test_models", "models", model_name1, NULL);
+  gchar *image_path = g_build_filename (
+      root_path, "tests", "test_models", "data", data_name, NULL);
+  gchar *pipeline_str;
+  GstElement *pipeline;
+  ASSERT_TRUE (g_file_test (model_path1, G_FILE_TEST_EXISTS));
+  ASSERT_TRUE (g_file_test (image_path, G_FILE_TEST_EXISTS));
+
+  pipeline_str = g_strdup_printf (
+      "filesrc location=%s ! pngdec ! videoscale ! imagefreeze ! videoconvert ! video/x-raw,format=RGB,framerate=0/1 ! tee name=t t. ! "
+      "queue ! videoscale ! videoconvert ! video/x-raw ! "
+      "tensor_converter ! tensor_filter name=filter1 framework=tensorflow-lite model=%s is-updatable=TRUE "
+      "shared-tensor-filter-key=%s ! tensor_sink name=sink1 t. ! "
+      "queue ! videoscale ! videoconvert ! video/x-raw,width=30,height=30 ! "
+      "tensor_converter ! tensor_filter name=filter2 framework=tensorflow-lite model=%s is-updatable=TRUE "
+      "shared-tensor-filter-key=%s ! tensor_sink name=sink2",
+      image_path, model_path1, shared_key, model_path1, shared_key);
+  g_free (root_path);
+  g_free (model_path1);
+  g_free (image_path);
+
+  pipeline = gst_parse_launch (pipeline_str, NULL);
+
+  EXPECT_NE (setPipelineStateSync (pipeline, GST_STATE_PLAYING, UNITTEST_STATECHANGE_TIMEOUT), 0);
```

Comment:
we will need to match the golden test data with this strategy, but I guess someone else could help this :)